be/src/util/bthreads/executor.cpp

```diff
@@ -12,13 +12,11 @@ ThreadPoolExecutor::~ThreadPoolExecutor() {
 
 int ThreadPoolExecutor::submit(void* (*fn)(void*), void* args) {
     Status st;
-    while (true) {
-        st = _thread_pool->submit_func([=]() { fn(args); });
-        if (!st.is_service_unavailable()) break;
-        LOG(INFO) << "async_delta_writer is busy, retry after " << _busy_sleep_ms << "ms";
-        bthread_usleep(_busy_sleep_ms * 1000);
-    }
-    LOG_IF(WARNING, !st.ok()) << st;
+    // We don't need to retry, because the queue length of the thread pool is 40960.
+    // If the queue is full, it probably means that there is a problem with the storage and the write is hung.
+    // If the error is not returned in time, all the writes in the cluster will be hung.
+    st = _thread_pool->submit_func([=]() { fn(args); });
+    LOG_IF(ERROR, !st.ok()) << st;
```

Comment:
The problem is if `ThreadPoolExecutor::submit()` returns -1 the whole process will crash.

Reply:
https://github.com/apache/incubator-brpc/blob/cf0aa330f625efd245914c406d37f1cf26c88679/src/bthread/execution_queue.cpp#L120

Reply:
Process exit seems not a bad idea, since we don't have measure to detect storage hang.
This problem has already occurred in the WeChat cluster. The storage hang of a single BE causes the entire cluster to fail to write, but the heartbeat is successful.

Reply:
https://github.com/apache/incubator-brpc/blob/731730da85f6af5c25012b4c83ab5bb371320cf8/docs/en/server.md#crash-after-printing-fatal-log

Reply:
> 

IMO, A better solution may be reject services.

Reply:
I use fatal log make process exit before we have other measure to detect, If it is really a storage problem may be a good choice let user notice quickly.